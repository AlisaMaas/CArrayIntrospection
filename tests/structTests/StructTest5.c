/**
* Ensure that we still catch
* struct accesses that are done via
* an array and a struct field access.
**/
struct foo {
	int a;
	double b;
	struct foo *c;
};

struct bar {
	struct foo first;
	int x;
	double y;
	struct bar *z;
};

void accessArrayStructElements(struct foo *f, struct bar *b) {
	for (int i = 0; i < 40; i++) {
		f[i].a = 9;
	}
	f[2].b = 2.2;
	b[200].first = f[0];
	b[0].x = 3;
	b[1].y = 3.3;
	b[4].z = b;
	f[0].c[4].a = 3;
	
}