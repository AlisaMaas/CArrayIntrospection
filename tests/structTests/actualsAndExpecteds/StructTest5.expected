struct.foo:
	element 0 accessed at a
	element 0 accessed at a10
	element 1 accessed at b2
	element 2 accessed at c
struct.bar:
	element 0 accessed at first
	element 1 accessed at x
	element 2 accessed at y
	element 3 accessed at z
